### PR TITLE
[1ES] Add 1ES pipeline files

### PR DESCRIPTION
**Pull request by jviau**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [ ] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR -- TODO -- blocked
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

This PR introduces new CI files for a new build pipeline.

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -40,10 +40,6 @@
       <Message Importance="high" Text="$(Version)" />
   </Target>
 
-  <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All"/>
```
Comment (jviau) [inline build/common.props:44]: This is part of the .NET8 SDK - no longer needed

File: eng/ci/integration-tests.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+variables:
+  DOTNET_NOLOGO: 1
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+  DOTNET_CLI_TELEMETRY_OPTOUT: 1
+
+pr:
+  branches:
+    include:
+    - dev
```
Comment (liliankasem) [inline eng/ci/integration-tests.yml:9]: do we not want to include the `in-proc` branch too?

File: eng/ci/integration-tests.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      image: 1es-windows-2022 
+      os: windows
+
+    stages:
+    - stage: Test
+      jobs:
+
+      - template: /eng/ci/templates/jobs/initialize-pipeline.yml@self
+      - template: /eng/ci/templates/official/jobs/run-non-e2e-tests.yml@self
+      - template: /eng/ci/templates/official/jobs/run-integration-tests.yml@self
```
Comment (liliankasem) [inline eng/ci/integration-tests.yml:36]: nit: We have a mix of E2E tests and integration tests, now that we're cleaning up, I would love for the integration tests to be split from E2E

File: eng/ci/templates/install-dotnet.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  displayName: Install .NET 3.1
+  inputs:
+    packageType: sdk
+    version: 3.1.x
+
+- task: UseDotNet@2 # Needed by our projects and CI steps
+  displayName: Install .NET 6
+  inputs:
+    packageType: sdk
+    version: 6.x
```
Comment (liliankasem) [inline eng/ci/templates/install-dotnet.yml:13]: Do we need to install 8.x too at any point?

File: eng/ci/integration-tests.yml (hunk 3 above)
Comment (jviau) [reply eng/ci/integration-tests.yml:36]: Sure - I'll take a look into that.

File: eng/ci/templates/install-dotnet.yml (hunk 4 above)
Comment (jviau) [reply eng/ci/templates/install-dotnet.yml:13]: The next task will do that - `Install .NET SDK from global.json`

**Review by liliankasem — APPROVED** [review]

File: eng/ci/integration-tests.yml (hunk 3 above)
Comment (jviau) [reply eng/ci/integration-tests.yml:36]: I think I'll leave this for another PR.

**Review by brettsam — APPROVED** [review]

**Pull request merged**